### PR TITLE
[SPARK-20218][Doc]'/applications/[app-id]/stages' in REST API,add description.

### PR DESCRIPTION
**Pull request by guoxiaolongzte**
## What changes were proposed in this pull request?

1. '/applications/[app-id]/stages' in rest api.status should add description '?status=[active|complete|pending|failed] list only stages in the state.'

Now the lack of this description, resulting in the use of this api do not know the use of the status through the brush stage list.

2.'/applications/[app-id]/stages/[stage-id]' in REST API,remove redundant description ‘?status=[active|complete|pending|failed] list only stages in the state.’.
Because only one stage is determined based on stage-id.

code:
  @GET
  def stageList(@QueryParam("status") statuses: JList[StageStatus]): Seq[StageData] = {
    val listener = ui.jobProgressListener
    val stageAndStatus = AllStagesResource.stagesAndStatus(ui)
    val adjStatuses = {
      if (statuses.isEmpty()) {
        Arrays.asList(StageStatus.values(): _*)
      } else {
        statuses
      }
    };

## How was this patch tested?

manual tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by srowen** [comment]
Why didn't you add this to your last PR? What other related changes do you want to make? Making several small closely related PRs is frowned on

**Comment by guoxiaolongzte** [comment]
Because I just use this API today to develop, only to find the problem.

**Comment by guoxiaolongzte** [comment]
Added one more

2.'/applications/[app-id]/stages/[stage-id]' in REST API,remove redundant description ‘?status=[active|complete|pending|failed] list only stages in the state.’.
Because only one stage is determined based on stage-id.

File: docs/monitoring.md [hunk 1]
```diff
@@ -299,12 +299,12 @@ can be identified by their `[attempt-id]`. In the API listed below, when running
   <tr>
     <td><code>/applications/[app-id]/stages</code></td>
     <td>A list of all stages for a given application.</td>
+    <br><code>?status=[active|complete|pending|failed]</code> list only stages in the state.
   </tr>
   <tr>
     <td><code>/applications/[app-id]/stages/[stage-id]</code></td>
     <td>
       A list of all attempts for the given stage.
-      <br><code>?status=[active|complete|pending|failed]</code> list only stages in the state.
```
Comment (srowen) [inline docs/monitoring.md]: Is it filtering stages? or attempts? I am not familiar with this but it wasn't obvious that this is redundant. Maybe you have tried this API locally and confirm this doesn't have effect?

Comment (guoxiaolongzte) [inline docs/monitoring.md]: It is filtering stages.
I have manual test this API locally and confirm this doesn't have effect.

**Comment by srowen** [comment]
Merged to master/2.1

**Pull request closed without merging**